src/platform/unix/mod.rs

```diff
@@ -472,7 +472,12 @@ impl OsIpcReceiverSet {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
             Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Err(ref e) if e.kind() != ErrorKind::Interrupted => {
```

Comment:
Another good point. This one is definitely difficult to reproduce as a unit test. I made my best attempt at it with b03e9db3a2ac87ec40dec3072eda0fa56e62a99c. Parts of it feel a bit hacky, like the "kill sled", but it does cause older versions to panic. It also caught a bug in the code! Hence the new changes to `OsIpcReceiverSet::select`